server/routerlicious/packages/lambdas/src/broadcaster/lambda.ts

```diff
@@ -42,42 +53,90 @@ export class BroadcasterLambda implements IPartitionLambda {
+
+                    const value = baseMessage as ISequencedOperationMessage;
+                    topic = `${value.tenantId}/${value.documentId}`;
+                    break;
+                }
+
+                case NackOperationType: {
+                    event = "nack";
+
+                    const value = baseMessage as INackMessage;
+                    topic = `client#${value.clientId}`;
+                    break;
+                }
 
-            if (topic && event) {
-                const value = baseMessage as INackMessage | ISequencedOperationMessage;
-
-                let pendingBatch = this.pending.get(topic);
-                if (!pendingBatch) {
-                    pendingBatch = {
-                        tenantId: value.tenantId,
-                        documentId: value.documentId,
-                        event,
-                        messages: [value.operation],
-                    };
-                    this.pending.set(topic, pendingBatch);
-                } else {
-                    pendingBatch.messages.push(value.operation);
+                case SignalOperationType: {
+                    event = "signal";
+
+                    const value = baseMessage as ITicketedSignalMessage;
+                    topic = `${value.tenantId}/${value.documentId}`;
+
+                    if (this.clientManager && value.operation) {
+                        const signalContent = JSON.parse(value.operation.content);
+                        const signalType: MessageType | undefined =
+                            typeof (signalContent.type) === "number" ? signalContent.type : undefined;
+                        switch (signalType) {
+                            case MessageType.ClientJoin:
+                                const signalClient: ISignalClient = signalContent.content;
+                                await this.clientManager.addClient(
+                                    value.tenantId,
+                                    value.documentId,
+                                    signalClient.clientId,
+                                    signalClient.client);
+                                break;
+
+                            case MessageType.ClientLeave:
+                                await this.clientManager.removeClient(
+                                    value.tenantId,
+                                    value.documentId,
+                                    signalContent.content);
+                                break;
+                        }
+                    }
+
+                    break;
                 }
+
+                default:
+                    // ignore unknown types
+                    continue;
+            }
+
+            const value = baseMessage as INackMessage | ISequencedOperationMessage | ITicketedSignalMessage;
+
+            if (this.serviceConfiguration.broadcaster.includeEventInMessageBatchName) {
+                topic += event;
+            }
```

Comment:
Whats the purpose of this flag? Is it related to this PR?

Reply:
This is required for things to work properly. This flag makes it separate ops & signals into separate batches. 

By default the topic name is `${tenantId}/${documentId}`. Without this flag, it ends up putting ops & signals in the same batch. And the "event" property for the batch would be based on when the batch is created. 

I think this could be a breaking change for FRS depending on how FRS is setup. So that's why I added a config for it.

Reply:
Ah makes sense. Yes not having the flag will break FRS/r11s since we create the channel in alfred: https://github.com/microsoft/FluidFramework/blob/main/server/routerlicious/packages/lambdas/src/alfred/index.ts#L241

This is a big change and its hard to say whether this will break r11s or not. @GaryWilber can you run docker and test basic scenarios? Otherwise @hedasilv or @znewton or @pradeepvairamani, can one of you help run these changes locally? Given the scope, I am pretty sure r11s and FRS will behave identically so testing r11s should be fine.

@GaryWilber I think you need to split the PR because proto-def package needs to be published first?

Reply:
I can't test this in docker as my work setup doesn't support it. I've been testing by running a version of Push with these changes included.
I was hoping the PR tests will be good enough to know if it's working or not.

Reply:
Do you mean the unit tests? They are probably not comprehensive enough.

Reply:
Was tested manually by @hedasilv